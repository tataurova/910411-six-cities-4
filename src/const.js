export const PLACES_COUNT = 312;

export const MapSettings = {
  city: [52.38333, 4.9],
  icon: {
    iconUrl: `img/pin.svg`,
    iconSize: [30, 30],
  },
  zoom: 12,
};
