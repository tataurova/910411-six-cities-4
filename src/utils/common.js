export const extend = (a, b) => {
  return Object.assign({}, a, b);
};

export const mockFunction = () => {
  // do nothing
};
