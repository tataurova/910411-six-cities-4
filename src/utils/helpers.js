export const getOfferInfo = (offers, offerId) => {
  const idNumber = Number(offerId);
  for (let key in offers) {
    if (offers.hasOwnProperty(key)) {
      const offer = offers[key].filter((item) => item.id === idNumber)[0];
      if (offer) {
        return {
          offer,
          nearOffers: offers[key],
        };
      }
    }
  }
  return {};
};
