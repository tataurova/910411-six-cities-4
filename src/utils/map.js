import leaflet from "leaflet";

export const addMarkersToMap = (offers, hoveredCardId, icon, activeIcon, map) => {
  const markers = [];
  offers.forEach((offer) => {
    const marker = leaflet.marker(offer.coordinates, {icon: offer.id === hoveredCardId ? activeIcon : icon});
    markers.push(marker);
  });
  const markersGroup = leaflet.layerGroup(markers).addTo(map);
  return markersGroup;
};
