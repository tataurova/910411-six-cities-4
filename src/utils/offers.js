export const getOfferInfo = (offers, offerId) => {
  const idNumber = Number(offerId);
  return {
    offer: offers.filter((item) => item.id === idNumber)[0],
    nearOffers: offers.filter((item) => item.id !== idNumber),
  };
  // for (let key in offers) {
  //   if (offers.hasOwnProperty(key)) {
  //     const offer = offers[key].filter((item) => item.id === idNumber)[0];
  //     if (offer) {
  //       return {
  //         offer,
  //         nearOffers: offers[key].filter((item) => item.id !== offer.id),
  //       };
  //     }
  //   }
  // }
  // return {};
};
