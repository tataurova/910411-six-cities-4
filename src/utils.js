import {SortType} from "./const.js";
import leaflet from "leaflet";

export const extend = (a, b) => {
  return Object.assign({}, a, b);
};

export const sortOffers = (sortType, offers) => {
  switch (sortType) {
    case SortType.DEFAULT:
      return offers;
    case SortType.PRICE_UP:
      return offers.sort((a, b) => a.price - b.price);
    case SortType.PRICE_DOWN:
      return offers.sort((a, b) => b.price - a.price);
    case SortType.RATING_DOWN:
      return offers.sort((a, b) => b.rating - a.rating);
  }
  return offers;
};

export const getSortTypeName = (sortType) => {
  switch (sortType) {
    case SortType.PRICE_UP:
      return `Price: low to high`;
    case SortType.PRICE_DOWN:
      return `Price: high to low`;
    case SortType.RATING_DOWN:
      return `Top rated first`;
  }
  return `Popular`;
};

export const getArrayOfOffers = (citiesOffers) => {
  let allOffers = [];
  Object.values(citiesOffers).forEach((item) => {
    allOffers = allOffers.concat(item);
  });
  return allOffers;
};

export const addMarkersToMap = (offers, hoveredCardId, icon, activeIcon, map) => {
  const markers = [];
  offers.forEach((offer) => {
    const marker = leaflet.marker(offer.coordinates, {icon: offer.id === hoveredCardId ? activeIcon : icon});
    markers.push(marker);
  });
  const markersGroup = leaflet.layerGroup(markers).addTo(map);
  return markersGroup;
};

export const getNearOffers = (offers, id, city) => {
  const numberId = Number(id);
  return offers.filter((offer) => offer.id !== numberId && offer.city === city);
};

export const findOfferById = (offers, id) => {
  return offers.filter((offer) => offer.id === id)[0];
};
