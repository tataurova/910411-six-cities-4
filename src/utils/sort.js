import {SortType} from "../const.js";

export const sortOffers = (sortType, offers) => {
  switch (sortType) {
    case SortType.DEFAULT:
      return offers;
    case SortType.PRICE_UP:
      return offers.sort((a, b) => a.price - b.price);
    case SortType.PRICE_DOWN:
      return offers.sort((a, b) => b.price - a.price);
    case SortType.RATING_DOWN:
      return offers.sort((a, b) => b.rating - a.rating);
  }
  return offers;
};

export const getSortTypeName = (sortType) => {
  switch (sortType) {
    case SortType.PRICE_UP:
      return `Price: low to high`;
    case SortType.PRICE_DOWN:
      return `Price: high to low`;
    case SortType.RATING_DOWN:
      return `Top rated first`;
  }
  return `Popular`;
};
