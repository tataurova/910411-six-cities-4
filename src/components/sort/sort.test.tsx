import * as React from "react";
import * as renderer from "react-test-renderer";
import Sort from "./sort";
import {SortType, DEFAULT_SORT_STATE} from "../../const";
import {mockFunction} from "../../utils/common";

describe(`<Sort />`, () => {
  it(`Should Sort render correctly`, () => {
    const tree = renderer
      .create(
          <Sort
            activeSortType = {SortType.DEFAULT}
            state = {DEFAULT_SORT_STATE}
            onSortTypeClick = {mockFunction}
            onChangeItem = {mockFunction}
          />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`Should Sort render correctly`, () => {
    const tree = renderer
      .create(
          <Sort
            activeSortType = {SortType.DEFAULT}
            state = {true}
            onSortTypeClick = {mockFunction}
            onChangeItem = {mockFunction}
          />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`Should Sort render correctly`, () => {
    const tree = renderer
      .create(
          <Sort
            activeSortType = {SortType.PRICE_UP}
            state = {true}
            onSortTypeClick = {mockFunction}
            onChangeItem = {mockFunction}
          />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
