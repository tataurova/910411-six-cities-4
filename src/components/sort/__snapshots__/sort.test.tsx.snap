// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Sort /> Should Sort render correctly 1`] = `
<form
  action="#"
  className="places__sorting"
  method="get"
>
  <span
    className="places__sorting-caption"
  >
    Sort by
  </span>
  <span
    className="places__sorting-type"
    onClick={[Function]}
    tabIndex={0}
  >
    Popular
    <svg
      className="places__sorting-arrow"
      height="4"
      width="7"
    >
      <use
        xlinkHref="#icon-arrow-select"
      />
    </svg>
  </span>
  <ul
    className="places__options places__options--custom "
  >
    <li
      className="places__option "
      data-sort="default"
      onClick={[Function]}
      tabIndex={0}
    >
      Popular
    </li>
    <li
      className="places__option "
      data-sort="price-up"
      onClick={[Function]}
      tabIndex={0}
    >
      Price: low to high
    </li>
    <li
      className="places__option "
      data-sort="price-down"
      onClick={[Function]}
      tabIndex={0}
    >
      Price: high to low
    </li>
    <li
      className="places__option "
      data-sort="rating-down"
      onClick={[Function]}
      tabIndex={0}
    >
      Top rated first
    </li>
  </ul>
</form>
`;

exports[`<Sort /> Should Sort render correctly 2`] = `
<form
  action="#"
  className="places__sorting"
  method="get"
>
  <span
    className="places__sorting-caption"
  >
    Sort by
  </span>
  <span
    className="places__sorting-type"
    onClick={[Function]}
    tabIndex={0}
  >
    Popular
    <svg
      className="places__sorting-arrow"
      height="4"
      width="7"
    >
      <use
        xlinkHref="#icon-arrow-select"
      />
    </svg>
  </span>
  <ul
    className="places__options places__options--custom places__options--opened"
  >
    <li
      className="places__option "
      data-sort="default"
      onClick={[Function]}
      tabIndex={0}
    >
      Popular
    </li>
    <li
      className="places__option "
      data-sort="price-up"
      onClick={[Function]}
      tabIndex={0}
    >
      Price: low to high
    </li>
    <li
      className="places__option "
      data-sort="price-down"
      onClick={[Function]}
      tabIndex={0}
    >
      Price: high to low
    </li>
    <li
      className="places__option "
      data-sort="rating-down"
      onClick={[Function]}
      tabIndex={0}
    >
      Top rated first
    </li>
  </ul>
</form>
`;

exports[`<Sort /> Should Sort render correctly 3`] = `
<form
  action="#"
  className="places__sorting"
  method="get"
>
  <span
    className="places__sorting-caption"
  >
    Sort by
  </span>
  <span
    className="places__sorting-type"
    onClick={[Function]}
    tabIndex={0}
  >
    Price: low to high
    <svg
      className="places__sorting-arrow"
      height="4"
      width="7"
    >
      <use
        xlinkHref="#icon-arrow-select"
      />
    </svg>
  </span>
  <ul
    className="places__options places__options--custom places__options--opened"
  >
    <li
      className="places__option "
      data-sort="default"
      onClick={[Function]}
      tabIndex={0}
    >
      Popular
    </li>
    <li
      className="places__option "
      data-sort="price-up"
      onClick={[Function]}
      tabIndex={0}
    >
      Price: low to high
    </li>
    <li
      className="places__option "
      data-sort="price-down"
      onClick={[Function]}
      tabIndex={0}
    >
      Price: high to low
    </li>
    <li
      className="places__option "
      data-sort="rating-down"
      onClick={[Function]}
      tabIndex={0}
    >
      Top rated first
    </li>
  </ul>
</form>
`;
